service/matching/matching_engine_test.go

```diff
@@ -2503,6 +2506,221 @@ func (s *matchingEngineSuite) TestDemotedMatch() {
 	task.finish(nil)
 }
 
+func TaskQueueMetricValidator(T *testing.T, capture *metricstest.Capture, FamilyCounterLength int, FamilyCounter float64, QueueCounterLength int, QueueCounter float64, QueuePartitionCounterLength int, QueuePartitionCounter float64) {
+	// checks the metrics according to the values passed in the parameters
+
+	snapshot := capture.Snapshot()
+	FamilyCounterRecordings := snapshot[metrics.LoadedTaskQueueFamilyCounter.Name()]
+	assert.Len(T, FamilyCounterRecordings, FamilyCounterLength)
+	counter, ok := FamilyCounterRecordings[FamilyCounterLength-1].Value.(float64)
+	assert.True(T, ok)
+	assert.Equal(T, FamilyCounter, counter)
```

Comment:
I don't think you need to do a type-check here since Equal takes interfaces anyway. you can just do
```suggestion
	assert.Equal(T, familyCounter, familyCounterRecordings[familyCounterLength-1].Value)
```